### ISSUE TITLE
initial load

### ISSUE THREAD

**Issue by istvanv-telenav** [mapbox/osm-comments-parser#45]
Initial load of changesets
When starting out with an empty database, there is an optimized way to load the initial backlog of changesets. Create an empty folder called csv in the project root and pass the option: initial=true changesetParser. After this command is run, run psql <db_name> < changesets/post_initial.sql to load the CSVs into the database. FIXME: this should be scripted

Could you tell me how would the command look like for passing the initial=true.
I can't seem to figure out where would I pass that option.
Thanks

**Comment by batpad** [comment]
This could probably be clearer in the documentation.

If you are importing using  the command line way:

```
node index.js changesets --filename=/path/to/xml/file --initial
```

If using as a node module:

```
var changesetParser = require('./changesets');
changesetParser({filename: '/path/to/changeset-xml', initial: true});
```

Hope this helps.

**Comment by istvanv-telenav** [comment]
It did. Thanks a lot :)

**Comment by istvanv-telenav** [comment]
I just have a short followup question. I've managed to import the csv into my db. I've got to tables populated (changesets and users). How do i get the rest of the tables filled with data such as stats or changeset_comments?

**Comment by batpad** [comment]
@istvanv-telenav strange - you should have 3 tables - `changesets`, `users` and `changeset_comments` -- did you get the `comments.csv` file at all? Any errors during the process?

The `stats` table works a bit differently - it is written to by the `changes` parser: https://github.com/mapbox/osm-comments-parser/tree/master/changes - it operates on the replication files with geometry changes, and does not have a back-fill process.

**Comment by istvanv-telenav** [comment]
no unfortunately I don't have a comments.csv file at all. no errors in the process. Maybe I've used the wrong file to parse? i tried with https://planet.openstreetmap.org/planet/changesets-latest.osm.bz2 should I use another?

**Comment by batpad** [comment]
@istvanv-telenav - ah, yes - you need the file that also contains the changeset discussions, which, I believe, would be: https://planet.openstreetmap.org/planet/discussions-latest.osm.bz2

If you do make a Pull Request to improve the README docs here, it would be very much appreciated.

**Comment by istvanv-telenav** [comment]
I will update it once I finish with the entire process. I've identified the timestamp of my initial import. What I  have to do now is apply the new replication files from https://planet.openstreetmap.org/replication/changesets/002/ using  **node index.js <notes|changesets> --filename=/path/to/xml/file**
The question is how do I keep it automatically updated?

**Comment by batpad** [comment]
> The question is how do I keep it automatically updated?

The frequency at which you pull replication files and run the parser is really up to you. The `osm-comments-parser` does not handle this. I do agree it might be nice to have some example code that shows how to run this at some frequency, but the gist is:

 - Check the `state` file at https://planet.openstreetmap.org/replication/changesets/state.yaml
 - Fetch latest changeset / newer changesets from when you had last fetched
 - Unzip each one and run the parser on it

And then you could run this as a `cron` job or however.

**Comment by istvanv-telenav** [comment]
I've managed to create a script and cron job to keep my database updated. with the files found at /replication/changesets/002
Unfortunately  my stats table is still empty! I am guessing that for it to be populated i would need 
https://planet.openstreetmap.org/replication/minute/002/983/ these files to be parsed.  Is that correct?

**Comment by batpad** [comment]
@istvanv-telenav that is correct - the `stats` table operates on the actual minutely geometry diffs and not the changeset metadata files.

**Comment by istvanv-telenav** [comment]
That means i would need another independent job that would run 
node changes/index.js changesets --filename=/path/to/203.osc   or what is this the correct command for it?
would it be able to parse Daily or hourly files instead of minutely?

**Comment by batpad** [comment]
It would be:

`node index.js changes --filename /path/to/203.osc` i.e. use the `index.js` in the parent folder, not inside `changes` -- the first argument `changes` tells it "what type of thing" to import.

You would probably need another independent cron job for this, yes. The `changes` parser should work for minutely, daily or hourly files, yes.

If you do feel like contributing back with documentation on setting up the cronjobs, etc. I'm sure it would be appreciated and possibly help others, would be happy to review.

**Comment by istvanv-telenav** [comment]
there's another issue i encountered
```
osm_comments/osm-comments-parser-master/changes/index.js:23                                             │HTTP request sent, awaiting response... 404 Not Found
        throw new Error('state file was not provided'); 
```
Could you give me some details regarding in what should the state file contain and where to place it.
 I've  tried adding a state file  https://planet.openstreetmap.org/replication/hour/000/049/904.state.txt 
into the parent folder and inside changes, but it doesn't seem to work. Does this file need to be renamed?
I assume the initial state file would contain the sequence id from where it should start parsing the data.
Thank you

**Comment by batpad** [comment]
@istvanv-telenav it seems to need an option like `--statefile` that points to a statefile with a replication id.

I'm a bit fuzzy on the details here -- @geohacker - ping - are you able to clarify what the `statefile` is expected to be here in the options passed to the `changes` parser, and how it would be passed?

**Comment by istvanv-telenav** [comment]
@batpad  I think i figured it out --statefile=osm_comments/osm-comments-parser-master/268.state.txt  but i think it only works on the **minutly diff** because the contents of the statefile differ from the hourly/daily replication files .

```
**minutely diff**
#Thu May 24 08:29:03 UTC 2018
txnMaxQueried=1625728980
sequenceNumber=2984268
timestamp=2018-05-24T08\:29\:02Z
txnReadyList=
txnMax=1625728980
txnActiveList=1625728822  
```
```
**hour/daily diff** 
#Thu May 24 08:02:09 UTC 2018
sequenceNumber=49921
timestamp=2018-05-24T08\:00\:00Z
```

**Comment by geohacker** [comment]
@istvanv-telenav yeah I think we've only tested for minutely replication files. But it shouldn't be hard to support hourly or daily - we only need the sequence number to start processing. We'd love a pull request!
